spinalcordtoolbox/image.py

```diff
@@ -970,26 +970,15 @@ def get_dimension(im_file, verbose=1):
     :param: im_file: Image or nibabel object
     :return: nx, ny, nz, nt, px, py, pz, pt
     """
-    # initialization
-    nx, ny, nz, nt, px, py, pz, pt = 1, 1, 1, 1, 1, 1, 1, 1
-    if type(im_file) is nib.nifti1.Nifti1Image:
+    if isinstance(im_file, (nib.nifti1.Nifti1Image, Image)):
         header = im_file.header
-    elif type(im_file) is Image:
-        header = im_file.hdr
     else:
         header = None
```

Comment:
Hmm, it looks like the `header is None` case isn't handled below, and would crash when taking `header.get_data_shape()` and `header.get_zooms()`.

Actually, this case _should_ crash? But maybe it should be with a `TypeError` instead (rather than an implicit `AttributeError`).